deploy/kubevirt-hostpath-security-constraints-csi.yaml

```diff
@@ -16,6 +16,8 @@ fsGroup:
   type: RunAsAny
 supplementalGroups:
   type: RunAsAny
-allowHostDirVolumePlugin: false
+allowHostDirVolumePlugin: true
+readOnlyRootFilesystem: false
```

Comment:
Are you sure these last two are needed? The operator doesn't set them.

Reply:
I think so, if you look at the daemonSet containers, they make use hostPath volumes here, https://github.com/kubevirt/hostpath-provisioner/blob/9a52f4b99614c1aa5b442db58e56fa020a955431/deploy/csi/csi-kubevirt-hostpath-provisioner.yaml#L222

and i think since it needs to create a directory per pv inside that mounted dir, the `readOnlyRootFilesystem` is also required to be set to `false`

Reply:
So readOnlyRootFilesystem is needed if you want to write to the root filesystem. This is a very bad idea since the OS is on that filesystem. In general you should have a separate partition or disk where you will create the directories for hpp.

The `allowHostDirVolumePlugin: true` is only needed in the context of the legacy provisioner which created actual hostpath volumes. This CSI provisioner creates CSI volumes, and does not require this special permission.

Reply:
> So readOnlyRootFilesystem is needed if you want to write to the root filesystem. This is a very bad idea since the OS is on that filesystem. In general you should have a separate partition or disk where you will create the directories for hpp.

right, we are setting this to `false` now

> The allowHostDirVolumePlugin: true is only needed in the context of the legacy provisioner which created actual hostpath volumes. This CSI provisioner creates CSI volumes, and does not require this special permission.

in our (CRC) case i think we need this option but will test if it work with `allowHostDirVolumePlugin: false`, anyways closing this PR as these changes are specific to CRC 's use-case and doesn't need to be included here, thanks!